src/test/java/com/commercetools/sync/products/ProductSyncTest.java

```diff
@@ -356,13 +364,263 @@ void sync_WithEmptyAttributeMetaDataMap_ShouldCallErrorCallback() {
+                mockProductTypeService, mockCategoryService, mock(TypeService.class),
+                mock(ChannelService.class), mock(CustomerGroupService.class), mock(TaxCategoryService.class),
+                mock(StateService.class),
+                mock(UnresolvedReferencesService.class),
+                mock(CustomObjectService.class));
+
+        final ProductSyncStatistics syncStatistics = productSync
+                .sync(singletonList(productDraft))
+                .toCompletableFuture()
+                .join();
+
+        // assertion
+        assertThat(syncStatistics).hasValues(1, 0, 1, 0, 0);
+        assertThat(exceptions).isEmpty();
+        assertThat(errorMessages).isEmpty();
+        assertThat(warningCallBackMessages).isEmpty();
+    }
+
+    @Test
+    void sync_withChangedProductButConcurrentModificationExceptionAndFailedFetchOnRetry_shouldFailedUpdateProduct() {
+        // preparation
+
+        final List<String> errorMessages = new ArrayList<>();
+        final List<Throwable> exceptions = new ArrayList<>();
+        final List<String> warningCallBackMessages = new ArrayList<>();
+
+        final ProductSyncOptions syncOptions = ProductSyncOptionsBuilder
+                .of(mock(SphereClient.class))
+                .errorCallback((exception, oldResource, newResource, updateActions) -> {
+                    errorMessages.add(exception.getMessage());
+                    exceptions.add(exception.getCause());
+                })
+                .warningCallback((exception, oldResource, newResource)
+                    -> warningCallBackMessages.add(exception.getMessage()))
+                .build();
+
+        final ProductDraft productDraft =
+                createProductDraftBuilder(PRODUCT_KEY_1_CHANGED_RESOURCE_PATH,
+                        ResourceIdentifier.ofKey("syncProductType"))
+                        .taxCategory(null)
+                        .state(null)
+                        .build();
+
+        final ProductService mockProductService = buildMockProductServiceWithFailedFetchOnRetry(productDraft);
+        final ProductTypeService mockProductTypeService = buildMockProductTypeService();
+        final CategoryService mockCategoryService = buildMockCategoryService();
+
+        final ProductSync productSync = new ProductSync(spy(syncOptions), mockProductService,
+                mockProductTypeService, mockCategoryService, mock(TypeService.class),
+                mock(ChannelService.class), mock(CustomerGroupService.class), mock(TaxCategoryService.class),
+                mock(StateService.class),
+                mock(UnresolvedReferencesService.class),
+                mock(CustomObjectService.class));
+
+        final ProductSyncStatistics syncStatistics = productSync
+                .sync(singletonList(productDraft))
+                .toCompletableFuture()
+                .join();
+
+        // assertion
+
+        // Test and assertion
```

Comment:
Why is this part called `// Test and assertion`?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/622/commits/685236a17bdb6c7e2af5d34da44a37ddfcac47a1